### PR TITLE
Ncnews 11

### PR DESCRIPTION
**Pull request by tubez7**
Completed ncnews-11

### CODE REVIEW

File: controllers/comments-controllers.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+const { insertCommentById } = require("../models/comments-models");
+
+exports.postCommentById = (req, res, next) => {
+  //   console.log(req, "whole request object");
+
+  const { article_id: articleId } = req.params; //article_id on params
+  const { username } = req.body; //username on body
+  const { body: commentBody } = req.body; // body on body
```
Comment (dav3rid) [inline controllers/comments-controllers.js:8]: You'd be able to destructure multiple properties from the same object on the same line eg.
```js
const { username, body: commentBody } = req.body;
```

File: controllers/comments-controllers.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+  insertCommentById(commentBody, articleId, username)
+    .then((comment) => {
+      res.status(201).send({ comment });
+    })
+    .catch((err) => {
+      if (err.constraint === "comments_article_id_fkey") {
+        res.status(404).send({ msg: `article ${articleId} not found` });
+      } else {
+        next(err);
```
Comment (dav3rid) [inline controllers/comments-controllers.js:18]: You can simply hardcode `next(err)` in your catch block here. This way you can defer any error handling to your error handling middleware functions.
You'd be welcome to use the same logic, but extracted elsewhere. Your controllers could quickly get bloated if there are more errors to consider at a later point

File: __tests__/app.test.js [hunk 3]
```diff
@@ -262,25 +262,79 @@ describe("GET request on /api/users", () => {
...
+  test("should respond with 400 bad request - INVALID USERNAME when article_id is not a number", () => {
+    const articleId = 1;
+    const comment = { username: "INVALID_ID", body: "test_body" };
+    return request(app)
+      .post(`/api/articles/${articleId}/comments`)
+      .send(comment)
+      .expect(400)
+      .then((res) => {
+        expect(res.body.msg).toBe("bad request - INVALID USERNAME");
+      });
```
Comment (dav3rid) [inline __tests__/app.test.js:325]: Nice test here! I would however suggest a 404 for this one. Also, the test description is a little misleading here

Comment (tubez7) [reply __tests__/app.test.js:325]: Resolved as per feedback

File: controllers/comments-controllers.js (hunk 1 above)
Comment (tubez7) [reply controllers/comments-controllers.js]: Updated as per feedback.

**Pull request merged**